subsys/net/ip/net_context.c

```diff
@@ -281,15 +270,14 @@ int net_context_get(sa_family_t family,
 		contexts[i].options.sndtimeo = K_FOREVER;
 #endif
 
-		if (IS_ENABLED(CONFIG_NET_IPV6) ||
-		    IS_ENABLED(CONFIG_NET_IPV4)) {
-			(void)memset(&contexts[i].remote, 0,
-				     sizeof(struct sockaddr));
-			(void)memset(&contexts[i].local, 0,
-				     sizeof(struct sockaddr_ptr));
-		}
+#if defined(CONFIG_NET_IP)
```

Comment:
I understand that using `IS_ENABLED()` with switch/case statements is problematic, but here? Why changing it, the compiler should rule out that code anyway?

Reply:
It's certainly a question of taste, therefore I reverted the change but my personal arguments are:
1) IMHO It's more obvious what is intended if you use simple preprocessor code without clever non-standard constructs unless such constructs provide an advantage that could not otherwise be achieved.
2) Code that is intended to always be removed should IMHO be eliminated by the preprocessor not by the optimizer. This prevents unintended side effects (e.g. flash/memory waste) when optimization is voluntarily disabled (e.g. to improve debugging experience).
3) Only #if / #endif has full IDE support. Most IDEs will dim code that is eliminated by the preprocessor, improve code navigation, etc. These advantages are lost when using non-standard constructs.

Just to let you know what I was thinking and to answer your question, not to start any argument of course!